crates/cheatnet/src/execution/syscalls/cairo0_cheatable_syscall_handler.rs

```diff
@@ -0,0 +1,286 @@
+use crate::state::CheatnetState;
+use blockifier::execution::common_hints::HintExecutionResult;
+use blockifier::execution::deprecated_syscalls::hint_processor::{
+    DeprecatedSyscallExecutionError, DeprecatedSyscallHintProcessor,
+};
+use blockifier::execution::deprecated_syscalls::{
+    CallContractRequest, DeprecatedSyscallResult, DeprecatedSyscallSelector, EmptyRequest,
+    GetBlockNumberResponse, GetBlockTimestampResponse, GetContractAddressResponse,
+    LibraryCallRequest, SyscallRequest, SyscallResponse, WriteResponseResult,
+};
+use blockifier::execution::execution_utils::{write_maybe_relocatable, ReadOnlySegment};
+use blockifier::execution::hint_code;
+
+use crate::execution::calls::cairo0_calls::execute_library_call;
+use crate::execution::syscalls::stark_felt_from_ptr_immutable;
+use cairo_felt::Felt252;
+use cairo_vm::hint_processor::builtin_hint_processor::builtin_hint_processor_definition::HintProcessorData;
+use cairo_vm::hint_processor::hint_processor_definition::{HintProcessorLogic, HintReference};
+use cairo_vm::serde::deserialize_program::ApTracking;
+use cairo_vm::types::exec_scope::ExecutionScopes;
+use cairo_vm::types::relocatable::Relocatable;
+use cairo_vm::vm::errors::hint_errors::HintError;
+use cairo_vm::vm::runners::cairo_runner::{ResourceTracker, RunResources};
+use cairo_vm::vm::vm_core::VirtualMachine;
+use num_traits::ToPrimitive;
+use starknet_api::block::{BlockNumber, BlockTimestamp};
+use starknet_api::core::ContractAddress;
+use std::any::Any;
+use std::collections::HashMap;
+
+#[derive(Debug)]
+// crates/blockifier/src/execution/syscalls/mod.rs:127 (SingleSegmentResponse)
+// It is created here because fields in the original structure are private
+// so we cannot create it in call_contract_syscall
+pub struct SingleSegmentResponse {
+    pub(crate) segment: ReadOnlySegment,
+}
+// crates/blockifier/src/execution/syscalls/mod.rs:131 (SyscallResponse for SingleSegmentResponse)
+impl SyscallResponse for SingleSegmentResponse {
+    fn write(self, vm: &mut VirtualMachine, ptr: &mut Relocatable) -> WriteResponseResult {
+        write_maybe_relocatable(vm, ptr, self.segment.length)?;
+        write_maybe_relocatable(vm, ptr, self.segment.start_ptr)?;
+        Ok(())
+    }
+}
+
+pub struct Cairo0CheatableSyscallHandler<'a> {
```

Comment:
Please use module name Cairo0 or Deprecated<sth> instead of prefixing everything